src/integrationtests/java/com/aws/greengrass/integrationtests/BridgePerformanceTest.java

```diff
@@ -0,0 +1,88 @@
+import com.aws.greengrass.mqtt.bridge.model.MqttMessage;
+import com.aws.greengrass.util.Pair;
+import org.junit.jupiter.api.AfterEach;
+
+import java.nio.charset.StandardCharsets;
+import java.time.Duration;
+import java.util.Collections;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.CountDownLatch;
+import java.util.concurrent.Executors;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.atomic.AtomicInteger;
+import java.util.function.Consumer;
+
+import static com.aws.greengrass.testcommons.testutilities.TestUtils.asyncAssertOnConsumer;
+import static com.github.grantwest.eventually.EventuallyLambdaMatcher.eventuallyEval;
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.is;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+@BridgeIntegrationTest
+public class BridgePerformanceTest {
+    ScheduledExecutorService ses = Executors.newScheduledThreadPool(1);
+
+    BridgeIntegrationTestContext context;
+
+    @AfterEach
+    void tearDown() {
+        ses.shutdownNow();
+    }
+
+    @TestWithMqtt5Broker
+    @WithKernel("mqtt5_local_to_local.yaml")
+    void GIVEN_local_to_local_bridge_WHEN_messages_sent_at_tps_THEN_messages_bridged(Broker broker) throws Exception {
+        AtomicInteger numMessagesBridged = new AtomicInteger();
+        AtomicInteger failedPublishes = new AtomicInteger();
+        AtomicInteger expectedPublishes = new AtomicInteger();
+
+        Pair<CompletableFuture<Void>, Consumer<MqttMessage>> cb
+                = asyncAssertOnConsumer(p -> numMessagesBridged.incrementAndGet());
+        context.getLocalV5Client().updateSubscriptions(
+                Collections.singleton("topic/toLocal"), cb.getRight());
+
+        int tps = 100;
+        Duration testDuration = Duration.ofSeconds(20);
+
+        CountDownLatch testFinished = new CountDownLatch(1);
+
+        ses.schedule(testFinished::countDown, testDuration.toMillis(), TimeUnit.MILLISECONDS);
+        ses.scheduleAtFixedRate(() -> {
+            try {
+                expectedPublishes.incrementAndGet();
+                context.getLocalV5Client().publish(
+                        MqttMessage.builder()
+                                .topic("topic/toLocal")
+                                .payload("message".getBytes(StandardCharsets.UTF_8))
+                                .build());
+            } catch (MessageClientException ignore) {
+                failedPublishes.incrementAndGet();
+            }
+        }, 0L, (long) (TimeUnit.SECONDS.toMicros(1) / (double) tps), TimeUnit.MICROSECONDS);
+
+        testFinished.await();
+
+        ses.shutdown();
+        assertTrue(ses.awaitTermination(5L, TimeUnit.SECONDS));
+
+        assertThat("messages bridged", expectedPublishes::get, eventuallyEval(is(numMessagesBridged.get())));
```

Comment:
this doesn't really look right. If we expect the failed publishes to be 0, then you should be checking the number of messages bridged is exactly equal to a constant value (not an AtomicInteger)